modules/150-user-authn/hooks/discover_dex_ca_test.go

```diff
@@ -97,9 +96,74 @@ data:
 			f.ValuesSet("userAuthn.controlPlaneConfigurator.dexCustomCA", "testca")
 			f.RunHook()
 		})
-		It("Should add no ca for OIDC provide from config", func() {
+
+		It("Should add no ca for OIDC provider from config", func() {
 			Expect(f).To(ExecuteSuccessfully())
 			Expect(f.ValuesGet("userAuthn.internal.discoveredDexCA").String()).To(Equal("testca"))
 		})
 	})
+
+	Context("Checking for CertManager and CustomcCertificate cases with secrets", func() {
+		BeforeEach(func() {
+			f.ValuesSet("userAuthn.controlPlaneConfigurator.dexCAMode", "FromIngressSecret")
+			f.BindingContexts.Set(f.KubeStateSetAndWaitForBindingContexts("", 0))
+			f.RunHook()
+		})
+
+		It("Should run", func() {
+			Expect(f).To(ExecuteSuccessfully())
+		})
+
+		Context("Matching with CertManager and ingress-tls", func() {
+			BeforeEach(func() {
+				f.ValuesSet("userAuthn.https.mode", "CertManager")
+				f.BindingContexts.Set(f.KubeStateSetAndWaitForBindingContexts(`
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ingress-tls
+  namespace: d8-user-authn
+data:
+  tls.crt: dGVzdA==
+`, 2))
+				f.RunHook()
+			})
+
+			It("Should add tls.crt for OIDC provider", func() {
+				Expect(f).To(ExecuteSuccessfully())
+				Expect(f.ValuesGet("userAuthn.internal.discoveredDexCA").String()).To(Equal("test"))
+			})
+
+			It("Should check non-existing ingress-tls-customcertificate secret", func() {
+				Expect(f).To(ExecuteSuccessfully())
+				Expect(f.KubernetesResource("Secret", "d8-user-authn", "ingress-tls-customcertificate").Exists()).To(BeFalse())
+			})
+		})
+
+		Context("Matching with CustomCertificate and ingress-tls-customcertificate", func() {
+			BeforeEach(func() {
+				f.ValuesSet("userAuthn.https.mode", "CustomCertificate")
+				f.BindingContexts.Set(f.KubeStateSetAndWaitForBindingContexts(`
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ingress-tls-customcertificate
+  namespace: d8-user-authn
+data:
+  tls.crt: dGVzdGNh
+`, 2))
+				f.RunHook()
+			})
```

Comment:
We were talking about two tests, like:
```
		Context("xxxx", func() {
			BeforeEach(func() {
				f.ValuesSet("userAuthn.https.mode", "CustomCertificate")
				f.BindingContexts.Set(f.KubeStateSetAndWaitForBindingContexts(`
apiVersion: v1
kind: Secret
metadata:
  name: ingress-tls
  namespace: d8-user-authn
data:
  tls.crt: dGVzdGNh
`, 2))
				f.RunHook()
			})
       It("should generate an error", func() {})
       		Context("Matching with CustomCertificate and ingress-tls-customcertificate", func() {
			BeforeEach(func() {
				f.ValuesSet("userAuthn.https.mode", "CustomCertificate")
				f.BindingContexts.Set(f.KubeStateSetAndWaitForBindingContexts(`
apiVersion: v1
kind: Secret
metadata:
  name: ingress-tls-customcertificate
  namespace: d8-user-authn
data:
  tls.crt: YWFhYQo=
---
apiVersion: v1
kind: Secret
metadata:
  name: ingress-tls
  namespace: d8-user-authn
data:
  tls.crt: dGVzdGNh
`, 2))
				f.RunHook()
			})
    It("Should use only  ingress-tls-customcertificate secret")
```

Reply:
You can keep these tests. But we need to check, that hook respects only necessary secret